cdc/sink/codec/canal_flat.go

```diff
@@ -362,3 +441,146 @@ func (c *CanalFlatEventBatchEncoder) SetParams(params map[string]string) error {
 	}
 	return nil
 }
+
+// CanalFlatEventBatchDecoder decodes the byte into the original message.
+type CanalFlatEventBatchDecoder struct {
+	data                []byte
+	msg                 *MQMessage
+	enableTiDBExtension bool
+}
+
+func NewCanalFlatEventBatchDecoder(data []byte, enableTiDBExtension bool) (EventBatchDecoder, error) {
```

Comment:
Seems the function `NewCanalFlatEventBatchDecoder` never returns an error, but why do you define your signature like this?